controllers/sandbox_controller.go

```diff
@@ -438,27 +437,29 @@ func (r *SandboxReconciler) reconcilePVCs(ctx context.Context, sandbox *sandboxv
 		pvc := &corev1.PersistentVolumeClaim{}
 		pvcName := pvcTemplate.Name + "-" + sandbox.Name
 		err := r.Get(ctx, types.NamespacedName{Name: pvcName, Namespace: sandbox.Namespace}, pvc)
-		if err != nil {
-			if k8serrors.IsNotFound(err) {
-				log.Info("Creating a new PVC", "PVC.Namespace", sandbox.Namespace, "PVC.Name", pvcName)
-				pvc = &corev1.PersistentVolumeClaim{
-					ObjectMeta: metav1.ObjectMeta{
-						Name:      pvcName,
-						Namespace: sandbox.Namespace,
-					},
-					Spec: pvcTemplate.Spec,
-				}
-				if err := ctrl.SetControllerReference(sandbox, pvc, r.Scheme); err != nil {
-					return fmt.Errorf("SetControllerReference for PVC failed: %w", err)
-				}
-				if err := r.Create(ctx, pvc, client.FieldOwner(sandboxControllerFieldOwner)); err != nil {
-					log.Error(err, "Failed to create PVC", "PVC.Namespace", sandbox.Namespace, "PVC.Name", pvcName)
-					return err
-				}
-			} else {
-				log.Error(err, "Failed to get PVC")
-				return fmt.Errorf("PVC Get Failed: %w", err)
-			}
+		if err == nil {
+			continue
+		}
+
+		if !k8serrors.IsNotFound(err) {
+			log.Error(err, "Failed to get PVC")
+			return fmt.Errorf("PVC Get Failed: %w", err)
+		}
+
+		log.Info("Creating a new PVC", "PVC.Namespace", sandbox.Namespace, "PVC.Name", pvcName)
+		pvc = &corev1.PersistentVolumeClaim{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      pvcName,
+				Namespace: sandbox.Namespace,
+			},
+			Spec: pvcTemplate.Spec,
+		}
+		if err := ctrl.SetControllerReference(sandbox, pvc, r.Scheme); err != nil {
+			return fmt.Errorf("SetControllerReference for PVC failed: %w", err)
```

Comment:
When setting the controller reference, the error message `SetControllerReference for PVC failed` is helpful, but it would be even better to include the name of the PVC for which it failed.